### DISCUSSION TITLE
how to get the position of a resource with respect to an html div

### DISCUSSION THREAD

**Discussion by Pablituuu** [fabricjs/fabric.js#9659] [Q&A] [answered]
Sorry for the inexperience.

I was looking for information about this, in fact it shouldn't be very complicated I think.

I want to obtain the position of a resource with respect to a div.
This arose because I have the scroll option activated and when I scroll up or scroll down, the position of the canvas changes and with it all the resources, that is, if a resource is too far to the left that is not visible on the canvas, there is a way to How to get the position.

![image](https://github.com/fabricjs/fabric.js/assets/61519207/b7158932-bb1e-4bf9-b8de-1d13c2c63088)

As in the example in the image, the element is very far to the left and I want to obtain the upper left position of it, but it is not visible because I scrolled the canvas.

I would greatly appreciate your kind support from any colleague.

**Comment by ShaMan123** [comment, marked as answer]
This should work if you did not apply any css to the canvas element I believe
```

sendPointToPlane(a.getCenterPoint(), a.getViewportTransform())
  .add(new Point(canvas._offset.left, canvas._offset.top)
```

**Reply by ShaMan123** [reply]
To get the top left the easiest way would be

```

sendPointToPlane(a.getCoords()[0], a.getViewportTransform())
  .add(new Point(canvas._offset.left, canvas._offset.top)

```

**Reply by ShaMan123** [reply]
And if you wish to get the tl point of the bbox that bounds the object as following:

<img width="382" alt="Screenshot 2024-02-10 at 11 26 30" src="https://github.com/fabricjs/fabric.js/assets/34343793/c153cc19-9fd1-45a7-9ddd-38f383aac278">


```
conts bbox = makeBoundingBoxFromPoints(a.getCoords().map(p => sendPointToPlane(p, a.getViewportTransform()))
new Point(bbox.left, bbox.top) .add(new Point(canvas._offset.left, canvas._offset.top)

```
